core/string/ustring.cpp

```diff
@@ -1975,14 +1987,16 @@ CharString String::ascii(bool p_allow_extended) const {
 
 	CharString cs;
 	cs.resize(size());
+	char *cs_ptrw = cs.ptrw();
+	const char32_t *this_ptr = ptr();
 
 	for (int i = 0; i < size(); i++) {
-		char32_t c = operator[](i);
+		char32_t c = this_ptr[i];
 		if ((c <= 0x7f) || (c <= 0xff && p_allow_extended)) {
-			cs[i] = c;
+			cs_ptrw[i] = c;
 		} else {
 			print_unicode_error(vformat("Invalid unicode codepoint (%x), cannot represent as ASCII/Latin-1", (uint32_t)c));
-			cs[i] = 0x20; // ascii doesn't have a replacement character like unicode, 0x1a is sometimes used but is kinda arcane
+			cs_ptrw[i] = 0x20; // ascii doesn't have a replacement character like unicode, 0x1a is sometimes used but is kinda arcane.
```

Comment:
You may as well make this one sparkle
```suggestion
			cs_ptrw[i] = 0x20; // Ascii doesn't have a replacement character like unicode, 0x1a is sometimes used but is kinda arcane.
```

Reply:
It's an abbreviation so don't think it should be capitalized, unless I make it `ASCII`

Reply:
ASCII it is (but it is a nitpick).